### ISSUE TITLE
Only show deprecated fields tab if they contain values

### ISSUE THREAD

**Issue by tobiasdiez** [JabRef/jabref#8396]
### JabRef version

5.4 (latest release)

### Operating system

Windows

### Details on version and operating system

Windows 11

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

The deprecated fields tab is currently shown even though the fields don't contain any values.
In such a case, the tab shouldn't be shown.
![image](https://user-images.githubusercontent.com/5037600/148381023-d10e7c2b-33b1-45ca-83b2-91abaf35f6ac.png)

**Comment by kenleyshaw** [comment]
I am new to the Open source community. For one of my courses for college, we have to contribute to a project - this seems like an issue that I may be able to help with. Can you assign this to me please? I'll update along the way as I troubleshoot

**Comment by tobiasdiez** [comment]
Welcome to JabRef! To make it easier to get started with developing we have written some guides; please have a look at https://devdocs.jabref.org/contributing and https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace. If you encounter any issues along the way, feel free to ask here or in our [gitter chat](https://gitter.im/JabRef/jabref).

**Comment by wujastyk** [comment]
+1  Screen real-estate is precious, and this is just a waste.  Note that none of my entrytypes, not one, has an `archiveprefix` field.

![image](https://user-images.githubusercontent.com/762246/151857067-80cf961d-6c8f-408f-a326-216dea459cfc.png)

**Comment by samruan-yz** [comment]
Hi community! I wonder if anyone is working on this issue currently? If not, may I handle this since it seems like a great issue to help me to get familiar with the code base before starting GSoC!

**Comment by ThiloteE** [comment]
Hey, @samruan-yz looks like this issue is partly a duplicate of #4688. Ideally, the "optional fields", "deprecated fields" and "other fields" tabs should be merged to one tab. The meta-issue https://github.com/JabRef/jabref/issues/4686 keeps track of the broader changes to the UI of the entry editor.

Just recently, some [students from Adelaide have been asking](https://github.com/JabRef/jabref/issues/4688#issuecomment-1093047724) to work on #4688, but they also have asked to work on other issues as well and there has not been any pull request yet. Maybe you can work together with @a1819644?

If #4688 is taken into account, this issue here would change from not showing the "tab", but not showing the "fields".

As I see it, there are now two different issues:

    A. The merger of the three tabs
    B. Make the deprecated fields not show, unless they contain values


It would make sense to do `A` before `B`, because if you make the fields vanish, but not the tab, there would be an empty "deprecated fields" tab, which is also disturbing to the user.
@samruan-yz, if you want to start on `B` first, you can do it, of course, since the merger would be done by @a1819644. 👍

**Comment by ThiloteE** [comment]
As a general advice: check out https://github.com/JabRef/jabref/blob/main/CONTRIBUTING.md for a start. Also, https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace is a good start. Feel free to ask if you have any questions here on GitHub or also at JabRef's [Gitter](https://gitter.im/JabRef/jabref) chat.

Try to open a (draft) pull request early on, so that people can see you are working on the issue and so that they can see the direction the pull request is heading towards. This way, you will likely receive valuable feedback. It also allows people to work on the same issue at the same time by keeping track of other's efforts, which is crucial at avoiding merge conflicts.

**Comment by a1819644** [comment]
Yeah sure. no problem. I can start working on merger "A"
@ThiloteE  thanks for the advice
